cmd/abapAddonAssemblyKitCheckCVs.go

```diff
@@ -48,9 +48,24 @@ func runAbapAddonAssemblyKitCheckCVs(config *abapAddonAssemblyKitCheckCVsOptions
 		}
 		c.copyFieldsToRepo(&addonDescriptor.Repositories[i])
 	}
-	log.Entry().Info("Write the resolved versions to the CommonPipelineEnvironment")
-	toCPE, _ := json.Marshal(addonDescriptor)
-	cpe.abap.addonDescriptor = string(toCPE)
+
+	var addonDescriptorFromCPE abaputils.AddonDescriptor
+	json.Unmarshal([]byte(config.AddonDescriptor), &addonDescriptorFromCPE)
+	addonDescriptor = combineYAMLRepositoriesWithCPEProduct(addonDescriptor, addonDescriptorFromCPE)
+
+	// now Software Component Versions fields are valid, but maybe Product Version was checked before, so copy that part from CPE
+	// we don't care for errors
+	// scenario 1: config.AddonDescriptor is empty since checkCVs is the first step in the pipeline, then the empty result is fine anyway
+	// scenario 2: for some reason config.AddonDescriptor is corrupt - then we insert the valid data but delete the repositories which will ensure issue is found later on
+	addonDescriptorCPE, _ := abaputils.ConstructAddonDescriptorFromJSON([]byte(config.AddonDescriptor))
+	if len(addonDescriptorCPE.AddonProduct) == 0 {
+		log.Entry().Info("No Product Version information present yet in the addonDescriptor of CommonPipelineEnvironment")
+	} else {
+		log.Entry().Infof("Information for Product Version %s taken from addonDescriptor of CommonPipelineEnvironment", addonDescriptorCPE.AddonProduct)
+	}
+	addonDescriptorCPE.SetRepositories(addonDescriptor.Repositories)
```

Comment:
I don't understand this logic completely. Why are addonDescriptorCPE and addonDescriptorFromCPE required?
In line 66, the repositories are set for "addonDescriptorCPE", but the "addonDescriptor" is written to the CPE in line 67. Is "addonDecrptorCPE" used anywhere? Maybe, I'm missing something...

Reply:
I deleted the "zombie" logic